packages/@aws-cdk/aws-servicecatalog/README.md

```diff
@@ -141,3 +164,38 @@ const product = new servicecatalog.CloudFormationProduct(this, 'MyFirstProduct',
   ]
 });
 ```
+
+### Adding a product to a portfolio
+
+You add products to a portfolio to manage your resources at scale.  After adding a product to a portfolio,
+it creates a portfolio-product association, and will become visible from the portfolio side in both the console and service catalog CLI.
+A product can be added to multiple portfolios depending on your resource and organizational needs.
+
+```ts fixture=portfolio-product
+portfolio.addProduct(product);
+```
+
+## Constraints
+
+Constraints define governance mechanisms that allow you to manage permissions, notifications, and options related to actions end users can perform on products,
+Constraints are applied on a portfolio-product association.
+Using the cdk, if you do not explicitly associate a product to a portfolio and add a constraint, it will automatically add an association for you.
+
+There are rules around plurariliites of constraints for a portfolio and product. For example, you can only have one `TagUpdate`
+constraint applied to a portfolio-product association. 
+If a misconfigured constraint is added, `synth` will fail with an error message.
+
+Read more at [Service Catalog Constraints](https://docs.aws.amazon.com/servicecatalog/latest/adminguide/constraints.html).
+
+### Resource update constraint
+
+Resource update constraints allow or disallow end users to update tags on resources associated with an AWS Service Catalog product upon provisioning. 
+If tag updating is allowed, then new tags associated with the product or portfolio will be applied to provisioned resources during a provisioned product update.
+
+```ts fixture=portfolio-product
+portfolio.addProduct(product);
+
+portfolio.addResourceUpdateConstraint(product, {
```

Comment:
1. I don't quite get why is this a method on Portfolio. This is more of a Product concern, no? For example, this "resource update constraint" only makes sense for CloudFormation products, I think?
2. I liked the previous name better. Also, I feel like we're overloading the term "add" here too much. We have `Portfolio.addProduct()`, `Portfolio.addTagOptions()`, and `Portfolio. addResourceUpdateConstraint ()`, which all do completely different things.
    How about `portfolio.constrainTagUpdates()`?

Reply:
1. Goal with using the `association-manager` is to add bidirectional usage of these APIs, so that end users can use whichever "side" they are operating from more.  You might a portfolio of products that you want to apply the same constraint on, it's more clear to call that from the portfolio side over all the added products.

Reply:
2. `constrainTagUpdates` seems a little more confusing since it doesn't tell you exactly what is going on, before we just had `allowTagUpdates` which seemed clearer since its basically a binary option, and it can be switched off so that it can be tracked over time.

Reply:
> 1. Goal with using the `association-manager` is to add bidirectional usage of these APIs, so that end users can use whichever "side" they are operating from more.  You might a portfolio of products that you want to apply the same constraint on, it's more clear to call that from the portfolio side over all the added products.

Is that possible though? Are all constraints applicable to all Products?

Reply:
Constraints are all applied on a portfolio-product association, so the bidirectional aspect will always be there.

Reply:
I also went back and forth to some of our docs and I really think that `allowTagUpdates` is actually the best name here.  Because the functionality is disabled by default, most users are creating this constraint to allow tag updates, hence why the default is true.  The option to shut it off only comes from a governance perspective where they purposefully would change that from it being true to false.  "Constrain" felt confusing because it is still just a boolean of yes/no.

Reply:
sorry, thought I was introducing confusion before with the bidirectional thing.  But yes, constraints are applicable to all types of products.